bash_completions.go

```diff
@@ -72,7 +72,7 @@ __%[1]s_handle_reply()
             else
                 allflags=("${flags[*]} ${two_word_flags[*]}")
             fi
-            COMPREPLY=( $(compgen -W "${allflags[*]}" -- "$cur") )
+            mapfile -t COMPREPLY < <(compgen -W "${allflags[*]}" -- "$cur")
```

Comment:
`mapfile` is not present in all bash versions. Namely, the out-of-the-box bash for MacOS doesn't have it (it's at a 3.0 version, it looks like `mapfile` was introduced in 4.0).

`read -a COMPREPLY` looks like it'll do the trick

Reply:
I think i need a project where i can test this. Disabling escape characters using `-r` is required as well (shellcheck warning) and the IFS seems to be required as well for it to work. So it's probably one of:
```
# For bash 3.x+
array=()
while IFS='' read -r line; do array+=("$line"); done < <(mycommand)
```
or
```
# For bash
IFS=" " read -r -a array <<< "$(mycommand)"
```
Some tests `read -a` in a bash shell didn't give me the right result (or i simply used it wrong).

Reply:
Yeah it seems `read -a` really doesn't work as expected. I've tested this on multiple bash versions, and it seems to be the most consistent:
```
while read -s -d " " c ; do
  COMPREPLY+=("$c")
done < <(compgen -W "${allflags[*]}" -- "$cur")
```

Reply:
Does it need to be cleared first?: `COMPREPLY=()`

Reply:
Bash will automatically create the array variable if you start appending to it:
```
unset a
a+=("foo")
echo "${a[@]}"
foo
```
But it wouldn't hurt to make sure it's empty anyway

Reply:
Was just worried that it is defined before already, but should be fine then.
Just compiled https://github.com/gohugoio/hugo with it but seems there is still sth. wrong with the completion (https://gist.github.com/rsteube/7647a9a3456cdb102f0b88ba004d74f8).
Currently looking into it.

Reply:
`while IFS='' read -r c; do` works in my shell (5.0.7).
https://gist.github.com/rsteube/6bb3336adf902193997bba048094b870

Reply:
3.2.57 looks ok as well: https://hub.docker.com/r/albers/bash-completion-mac